### ISSUE TITLE
Config overrides for development

### ISSUE THREAD

**Issue by mightyguava** [square/misk#909]
Misk lacks a supported, straightforward way to override configuration in development. Some common overrides include:

* log level
* full request logging
* host/port/user/password for external dependencies
* serving port
* enable/disable caching

The dev currently can modify/create the `src/main/resources/<app_name>-development.yaml` file to achieve overrides, but this has the downside that the changes must be reverted/omitted when committing code.

With a little work, we could let the dev specify the path to an `overrides.yaml` somewhere on their filesystem. But having to modify a file to change a commonly overridden flag is still more overhead than we want, and the dev could end up with a bunch of `override-this-flag.yaml` and `override-that-flag.yaml`. The dev would also have to open the file to recall what the override actually is.

Ideally we stick to conventions and allow overrides via environment variables or flags. I.e. one possible approach with flags is that, in the following config:

```
  web:
    port: 8080
    idle_timeout: 30000

  http_clients:
    endpoints:
      web_proxy_action:
        url: http://localhost/
```

If we wanted to override `port`, we could add a flag like `java myapp.jar -- --web.port=8000`. Similarly, for the `web_proxy_action`, it could be `--http_clients.endpoints.web_proxy_action.url=http://127.0.0.1/`. Multiple flags can be provided to override multiple configurations.

**Comment by mmihic** [comment]
I know you say you don't want to use override files, but support for override files is already built into `MiskConfig`. `SkimConfig` specifies the override path for non-`DEVELOPMENT` environments; would be a 2 line change to also support override paths for `DEVELOPMENT` environments

**Comment by mightyguava** [comment]
Yeah! Adding support to Skim for a config override path flag would be a great start to this. I'm not seeing special casing where it specifies an override path for non-`DEVELOPMENT` environments though.
